examples/jumbo-frame/README.md

```diff
@@ -0,0 +1,101 @@
+
+For this example, we can use the `tracepath` utility to make sure jumbo frames are supported on our local machine.
+
+```console
+[ec2-user@ip-1-1-1-1 jumbo-frame]$ tracepath localhost
+ 1:  localhost                                             0.035ms reached
+     Resume: pmtu 65535 hops 1 back 1
+```
+The tracepath tool says something about `pmtu 65535`, which means that the Path Maximum Transmission Unit is 65,535 bytes. So 9,001 bytes should be no problem!
+
+Then we just need to configure the io provider that we pass into our quic endpoint and we are good to go. The full code is in the folder, but the most relevant snippet is reproduced below.
+```rust
+    let address: SocketAddr = "127.0.0.1:4433".parse()?;
+    let io = s2n_quic::provider::io::Default::builder()
+        .with_receive_address(address)?
+        // Specify that the endpoint should try to use frames up to 9001 bytes.
+        // This is the maximum mtu that most ec2 instances will support.
+        // https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/network_mtu.html
+        .with_max_mtu(9001)?
+        // It's wise to benchmark for your individual usecase, but for the high
+        // throughput scenarios that jumbo frames sometimes enable, it is wise
+        // to set larger buffers on the sockets.
+        .with_recv_buffer_size(12_000_000)?
+        .with_send_buffer_size(12_000_000)?
+        .build()?;
+```
+
+We can run the demo by spinning up two terminals. In the first terminal, start the server
+```console
+[ec2-user@ip-172-31-28-149 jumbo-frame]$ pwd
+/home/ec2-user/workspace/s2n-quic/examples/jumbo-frame
+[ec2-user@ip-172-31-28-149 jumbo-frame]$ cargo run --bin jumbo_server
+   Compiling rustls-provider v0.1.0 (/home/ec2-user/workspace/s2n-quic/examples/jumbo-frame)
+    Finished dev [unoptimized + debuginfo] target(s) in 3.23s
+     Running `target/debug/jumbo_server`
+Listening for a connection
+
+```
+
+In the other terminal, go ahead and start the client
+```console
+[ec2-user@ip-172-31-28-149 jumbo-frame]$ cargo run --bin jumbo_client
+   Compiling rustls-provider v0.1.0 (/home/ec2-user/workspace/s2n-quic/examples/jumbo-frame)
+    Finished dev [unoptimized + debuginfo] target(s) in 3.44s
+     Running `target/debug/jumbo_client`
+MtuUpdated { path_id: 0, mtu: 1200, cause: NewPath }
+MtuUpdated { path_id: 0, mtu: 1472, cause: ProbeAcknowledged }
+MtuUpdated { path_id: 0, mtu: 5222, cause: ProbeAcknowledged }
+MtuUpdated { path_id: 0, mtu: 7097, cause: ProbeAcknowledged }
+MtuUpdated { path_id: 0, mtu: 8035, cause: ProbeAcknowledged }
+MtuUpdated { path_id: 0, mtu: 8504, cause: ProbeAcknowledged }
+MtuUpdated { path_id: 0, mtu: 8738, cause: ProbeAcknowledged }
+MtuUpdated { path_id: 0, mtu: 8855, cause: ProbeAcknowledged }
+MtuUpdated { path_id: 0, mtu: 8914, cause: ProbeAcknowledged }
+MtuUpdated { path_id: 0, mtu: 8943, cause: ProbeAcknowledged }
```

Comment:
Not blocking this PR but I'm wondering if this is the best strategy... That's quite a few probes that we're sending. For networks that you know do support jumbo frames, like EC2, it might be more optimal to try and jump to the highest packet size.

Reply:
Attempting to summarize offline conversation here.

Starting the probe at the lower end and going upwards was chosen to keep the probing overhead low on networks when
- users were sending small packet &&
- users enabled jumbo frames, but they were not supported on the network

That case performs particularly poorly because
- For each acked packet, the congestion window grows by packet_size
- so the loss of a 9,000 byte probe slows down the growth of the congestion window

That being said, user's that are sure of their network topology and MTU would gain some benefit by immediately probing for the maximum mtu, so we should figure out a way to unblock them, although that work is outside the scope of this PR.